bull/pdl-queue/consumer.js

```diff
@@ -1,175 +1,221 @@
-    byteLength = Number(byteLength + no_of_pages * 16) //No. of pages * 16
-    return [zip, byteLength]
+  var zip = new JSZip();
+  title = title.replace(/ /g, "_");
+  var img = zip.folder(`${title}_images`);
+  let temp_pages = no_of_pages;
+  var download_image = async function(uri, filename) {
+    await rp({
+      method: "GET",
+      uri,
+      encoding: null,
+      transform: function(body, response) {
+        return { headers: response.headers, data: body };
+      }
+    })
+      .then(async function(body) {
+        if (/image/.test(body.headers["content-type"])) {
+          var data = new Buffer(body.data);
+          img.file(filename, data.toString("base64"), { base64: true });
+        }
+      })
+      .catch(function(err) {
+        --no_of_pages;
+      });
+  };
+  for (let i = 1; i <= temp_pages; ++i) {
+    const str = `http://www.punjabdigilib.org/images?ID=${id}&page=${i}&pagetype=null&Searched=W3GX`;
+    await download_image(str, `${title}_${i}.jpeg`);
+    console.log("Image " + i + " zipped.");
+  }
+  let { byteLength } = await zip.generateAsync({ type: "nodebuffer" });
+  byteLength = Number(byteLength + no_of_pages * 16); //No. of pages * 16
+  return [zip, byteLength];
 }
 
 function setHeaders(metadata, byteLength, title) {
-    let headers = {};
-    headers["Authorization"] = `LOW ${process.env.access_key}:${process.env.secret_key}`
-    headers["Content-type"] = "application/zip"
-    headers["Content-length"] = byteLength
-    headers["X-Amz-Auto-Make-Bucket"] = 1
-    headers["X-Archive-meta-collection"] = "opensource"
-    headers["X-Archive-Ignore-Preexisting-Bucket"] = 1
-    headers["X-archive-meta-identifier"] = title
-    headers["X-archive-meta-mediatype"] = "texts"
-    headers["X-archive-meta-uploader"] = "bub.wikimedia@gmail.com" //To be added
-    headers["X-archive-meta-contributor"] = "Panjab Digital Library" //To be added
-    headers["X-archive-meta-betterpdf"] = true //To be added
-    headers["X-archive-meta-external-identifier"] = `urn:pdl:${metadata['bookID']}:${metadata['categoryID']}` //To be added
-    for (var key in metadata) {
-        let meta_key = key.trim().replace(/ /g, "-").toLowerCase()
-        headers[`X-archive-meta-${meta_key}`] = metadata[key];
-    }
-    headers['X-archive-meta-title'] = metadata['title'];
-    return headers;
+  let headers = {};
+  headers[
+    "Authorization"
+  ] = `LOW ${process.env.access_key}:${process.env.secret_key}`;
+  headers["Content-type"] = "application/zip";
+  headers["Content-length"] = byteLength;
+  headers["X-Amz-Auto-Make-Bucket"] = 1;
+  headers["X-Archive-meta-collection"] = "opensource";
+  headers["X-Archive-Ignore-Preexisting-Bucket"] = 1;
+  headers["X-archive-meta-identifier"] = title;
+  headers["X-archive-meta-mediatype"] = "texts";
+  headers["X-archive-meta-uploader"] = "bub.wikimedia@gmail.com"; //To be added
+  headers["X-archive-meta-contributor"] = "Punjab Digital Library"; //To be added
```

Comment:
It should be `Panjab` here as well. The changes should be made only where 'Panjab' is getting displayed, not while interacting with the APIs.